Source/Scene/ModelExperimental/ModelExperimentalAnimationCollection.js

```diff
@@ -0,0 +1,532 @@
+    if (animations[i].name === options.name) {
+      index = i;
+      break;
+    }
+  }
+
+  //>>includeStart('debug', pragmas.debug);
+  if (!defined(index)) {
+    throw new DeveloperError("options.name must be a valid animation name.");
+  }
+  //>>includeEnd('debug');
+
+  return addAnimation(this, animations[index], options);
+};
+
+/**
+ * Creates and adds animations with the specified initial properties to the collection
+ * for all animations in the model.
+ * <p>
+ * This raises the {@link ModelExperimentalAnimationCollection#animationAdded} event for each model so, for example, a UI can stay in sync.
+ * </p>
+ *
+ * @param {Object} [options] Object with the following properties:
+ * @param {JulianDate} [options.startTime] The scene time to start playing the animations. When this is <code>undefined</code>, the animations starts at the next frame.
+ * @param {Number} [options.delay=0.0] The delay, in seconds, from <code>startTime</code> to start playing.
+ * @param {JulianDate} [options.stopTime] The scene time to stop playing the animations. When this is <code>undefined</code>, the animations are played for its full duration.
+ * @param {Boolean} [options.removeOnStop=false] When <code>true</code>, the animations are removed after they stop playing.
+ * @param {Number} [options.multiplier=1.0] Values greater than <code>1.0</code> increase the speed that the animations play relative to the scene clock speed; values less than <code>1.0</code> decrease the speed.
+ * @param {Boolean} [options.reverse=false] When <code>true</code>, the animations are played in reverse.
+ * @param {ModelAnimationLoop} [options.loop=ModelAnimationLoop.NONE] Determines if and how the animations are looped.
+ * @returns {ModelExperimentalAnimation[]} An array of {@link ModelExperimentalAnimation} objects, one for each animation added to the collection.  If there are no glTF animations, the array is empty.
+ *
+ * @exception {DeveloperError} Animations are not loaded. Wait for the {@link ModelExperimental#readyPromise} to resolve.
+ * @exception {DeveloperError} options.multiplier must be greater than zero.
+ *
+ * @example
+ * model.activeAnimations.addAll({
+ *   multiplier : 0.5,                            // Play at half-speed
+ *   loop : Cesium.ModelAnimationLoop.REPEAT      // Loop the animations
+ * });
+ */
+ModelExperimentalAnimationCollection.prototype.addAll = function (options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+  const model = this._model;
+
+  //>>includeStart('debug', pragmas.debug);
+  if (!model.ready) {
+    throw new DeveloperError(
+      "Animations are not loaded.  Wait for Model.readyPromise to resolve."
+    );
+  }
+
+  if (defined(options.multiplier) && options.multiplier <= 0.0) {
+    throw new DeveloperError("options.multiplier must be greater than zero.");
+  }
+  //>>includeEnd('debug');
+
+  const animations = model.sceneGraph.components.animations;
+
+  const runtimeAnimations = [];
```

Comment:
Maybe rename this to `addedAnimations` to make it clearer that this is just the new animations that were added. I misread this as a local variable version of `this._runtimeAnimations`, which is not exactly true.